### DISCUSSION TITLE
What is the best tips or your suggestions regarding refreshing data in dataTable?

### DISCUSSION THREAD

**Discussion by anonymouse703** [livewire/livewire#5183] [1. Help]
In my previous project I used Bootstrap DataTable and I already solved on how to re-render the data in Bootstrap DataTable but what I notice is it's so lag when you have 100s of data in your query and you need ajax for its server side rendering but I don't want to do it since the purpose of livewire is to shorten the js code. 

So in my recent project I tried to used the power grid dataTable package and it's have the same problem on re-rendering the data in the dataTable after I store data and close the modal. I still need to refresh the page to display the new data.

So can I ask any recommendation and tips on how to work dataTable in livewire? The scenario of my new project is this:

I have 3 components
1. The parent component (for my list)
2. Modal for form
3. since I used powergrid dataTable it will generate another component for datable

The problem is $refresh is only refreshing the parent component but not the other component (dataTable) so every time I store another data I need to refresh the page for the new data to display.

I hope you have suggestions and tips about this or this will be given a solution soon.

Thank you and God bless

**Comment by Martleby** [comment]
Hello,

In Livewire you can use events and listeners to communicate between components on a page. When you store a new item you could raise an 'item-added' event which you can listen for on your table, and then refresh the table when it receives the event.

Have a read through the docs here: https://laravel-livewire.com/docs/2.x/events

Cheers,
Martin.
